WordPress/Classes/ViewRelated/Reader/ReaderDetailWebviewViewController.swift

```diff
@@ -0,0 +1,83 @@
+import UIKit
+
+protocol ReaderDetailView: class {
+    func render(_ post: ReaderPost)
+    func showError()
+}
+
+class ReaderDetailWebviewViewController: UIViewController, ReaderDetailView {
+    @IBOutlet weak var webView: WKWebView!
+
+    /// The coordinator, responsible for the logic
+    var coordinator: ReaderDetailCoordinator!
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        coordinator.start()
+    }
+
+    func render(_ post: ReaderPost) {
+        webView.loadHTMLString(post.contentForDisplay(), baseURL: nil)
+    }
+
+    func showError() {
+        // Show error
+    }
+
+    /// A View Controller that displays a Post content.
+    ///
+    /// Use this method to present content for the user.
+    /// - Parameter postID: a post identification
+    /// - Parameter siteID: a site identification
+    /// - Parameter isFeed: a Boolean indicating if the site is an external feed (not hosted at WPcom and not using Jetpack)
+    /// - Returns: A `ReaderDetailWebviewViewController` instance
+    @objc class func controllerWithPostID(_ postID: NSNumber, siteID: NSNumber, isFeed: Bool = false) -> ReaderDetailWebviewViewController {
+        let controller = ReaderDetailWebviewViewController.loadFromStoryboard()
+        let coordinator = ReaderDetailCoordinator(view: controller)
+        coordinator.set(postID: postID, siteID: siteID, isFeed: isFeed)
+        controller.coordinator = coordinator
+
+        return controller
+    }
+
+    /// A View Controller that displays a Post content.
+    ///
+    /// Use this method to present content for the user.
+    /// - Parameter url: an URL of the post.
+    /// - Returns: A `ReaderDetailWebviewViewController` instance
+    @objc class func controllerWithPostURL(_ url: URL) -> ReaderDetailWebviewViewController {
+        let controller = ReaderDetailWebviewViewController.loadFromStoryboard()
+
+        return controller
+    }
+
+    /// A View Controller that displays a Post content.
+    ///
+    /// Use this method to present content for the user.
+    /// - Parameter post: a Reader Post
+    /// - Returns: A `ReaderDetailWebviewViewController` instance
+    @objc class func controllerWithPost(_ post: ReaderPost) -> ReaderDetailWebviewViewController {
+        if post.sourceAttributionStyle() == .post &&
+            post.sourceAttribution.postID != nil &&
+            post.sourceAttribution.blogID != nil {
+            return ReaderDetailWebviewViewController.controllerWithPostID(post.sourceAttribution.postID!, siteID: post.sourceAttribution.blogID!)
+        } else if post.isCross() {
+            return ReaderDetailWebviewViewController.controllerWithPostID(post.crossPostMeta.postID, siteID: post.crossPostMeta.siteID)
+        } else {
```

Comment:
In what cases will this else happen?

Reply:
Based on my tests, most of the time. I'm still not sure what `sourceAttributionStyle` means, but I haven't created this code. I just adapted from here: https://github.com/wordpress-mobile/WordPress-iOS/blob/develop/WordPress/Classes/ViewRelated/Reader/ReaderStreamViewController.swift#L1551